CDCgov/prime-reportstream
# Implement proof of concept app insights logging using ThreadContext

This PR implement a proof of concept for azure app insights logging using log4j ThreadContext. According to azure documentation, ThreadContext data should get automatically sent into azure app insights. We are unsure if it means it will get sent as raw text or as searchable "customDimensions." We need to merge this PR to staging to find out. Even if this does not work the way we want in app insights, its still a helpful change that may help debugging easier and can be expanded at a later date.

Test Steps:
1. Send a report via RS API
2. Verify the following messages appear in local log:

```
[2023-01-17T17:04:53.640Z] 11:04:53.639 [pool-2-thread-48][71] INFO  gov.cdc.prime.router.azure.DatabaseAccess - Action history for action 'receive' has been recorded - {action_id=310, action_name=receive, duration=1118, end_time=2023-01-17T11:04:53.639755, sending_organization=simple_report, start_time=2023-01-17T11:04:52.521560, username=null}
```

## Changes
- Update `ActionHistory.kt` to include timestamp of when it is created
- Update `DatabaseAccess.saveActionHistoryToDb` to send a "complete" message with metadata via `ThreadContext`

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #7706

## To Be Done
- On staging: Verify Azure app insights includes the ThreadContext metadata as searchable customDimensions
